### PR TITLE
Rely on initial pthread_jit_write_protect_np state

### LINKED ISSUE #41991: OSX-Arm64 pthread_jit_write_protect_np(bool) does not have a sane initial value
Discovered while debugging the `Exceptions/ForeignThread/ForeignThreadExceptions/ForeignThreadExceptions.dll` test.

The test creates a new thread which invokes JIT managed code.
https://github.com/dotnet/runtime/blob/00d0dba14ed9927b22de224de94d812449529f17/src/tests/Exceptions/ForeignThread/ForeignThreadExceptionsNative.cpp#L49-L60

```
(lldb) bt
* thread #1, queue = 'com.apple.main-thread'
  * frame #0: 0x00000001d51db3d0 libsystem_kernel.dylib`__ulock_wait + 8
    frame #1: 0x00000001d528ec3c libsystem_pthread.dylib`_pthread_join + 452
    frame #2: 0x0000000102377ed8 libForeignThreadExceptionsNative.dylib`::InvokeCallbackOnNewThread(callback=(0x000000028003c210)) at ForeignThreadExceptionsNative.cpp:60:10
    frame #3: 0x000000028137ce34
    frame #4: 0x000000028137cbac
    frame #5: 0x000000028137c914
    frame #6: 0x000000010163f40c libcoreclr.dylib`CallDescrWorkerInternal at calldescrworkerarm64.S:71
```

The new thread is failing when it tries to execute the managed code.
```
* thread #9, stop reason = EXC_BAD_ACCESS (code=2, address=0x28003c210)
    frame #0: 0x000000028003c210
  * frame #1: 0x0000000102377e38 libForeignThreadExceptionsNative.dylib`InvokeCallbackUnix(callback=0x000000028003c210) at ForeignThreadExceptionsNative.cpp:31:5
    frame #2: 0x00000001d528d0f8 libsystem_pthread.dylib`_pthread_start + 320
(lldb) mem region 0x000000028003c210
[0x000000028003c000-0x0000000280040000) rwx
```

This is apparently occurring because the new thread does not default to `pthread_jit_write_protect_np(true)`.

I believe we should try to have Apple set a default which allows new threads to execute JIT code.

/cc @janvorli

### PR DESCRIPTION
**Pull request by sdmaclea**
Fixes #41991

### CODE REVIEW

File: src/coreclr/pal/src/map/virtual.cpp [hunk 1]
```diff
@@ -1765,8 +1765,8 @@ bool
 PAL_JITWriteEnableHolder::JITWriteEnable(bool writeEnable)
 {
     // Use a thread local to track per thread JIT Write enable state
-    // Initialize threads to start with MAP_JIT pages readable and executable (R-X) by default.
-    thread_local bool enabled = (pthread_jit_write_protect_np(1), false);
+    // Per Apple, new threads start with MAP_JIT pages readable and executable (R-X) by default.
```
Comment (jkoritzinsky) [inline src/coreclr/pal/src/map/virtual.cpp:1768]: Is there a place in Apple's documentation that we could put a link to for this?

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

File: src/coreclr/pal/src/map/virtual.cpp (hunk 1 above)
Comment (sdmaclea) [reply src/coreclr/pal/src/map/virtual.cpp:1768]: Apple said they were going to update `man pthread_jit_write_protect_np` but it is not currently documented.

**Comment by sdmaclea** [comment]
`baseservices\exceptions\Reports\baseservices.exceptions\sharedexceptions\emptystacktrace\OOMException01` is failing on every platform.  This is not related.

**Pull request merged**